django_setup_configuration/management/commands/generate_config_docs.py

```diff
@@ -0,0 +1,174 @@
+
+        if not (info := getattr(config_settings, "detailed_info", None)):
+            return
+
+        for key, value in info.items():
+            part = []
+            part.append(f"{'Variable':<20}{value['variable']}")
+            part.append(
+                f"{'Description':<20}{value['description'] or 'No description'}"
+            )
+            part.append(
+                f"{'Possible values':<20}"
+                f"{value.get('possible_values') or 'No information available'}"
+            )
+            part.append(
+                f"{'Default value':<20}{value.get('default_value') or 'No default'}"
+            )
+            result.append(part)
+
+    def get_detailed_info(
+        self,
+        config: ConfigSettings,
+        config_step,
+        related_steps: list,
+    ) -> list[list[str]]:
+        """
+        Get information about the configuration settings:
+            1. from model fields associated with the `ConfigSettings`
+            2. from information provided manually in the `ConfigSettings`
+            3. from information provided manually in the `ConfigSettings` of related
+               configuration steps
+        """
+        ret = []
+        for field in config.config_fields:
+            part = []
+            part.append(f"{'Variable':<20}{config.get_config_variable(field.name)}")
+            part.append(f"{'Setting':<20}{field.verbose_name}")
+            part.append(f"{'Description':<20}{field.description or 'No description'}")
+            part.append(f"{'Possible values':<20}{field.field_description}")
+            part.append(f"{'Default value':<20}{field.default_value}")
+            ret.append(part)
+
+        self._add_detailed_info(config, ret)
+
+        for step in related_steps:
+            self._add_detailed_info(step.config_settings, ret)
+
+        return ret
+
+    def format_display_name(self, display_name: str) -> str:
+        """Surround title with '=' to display as heading in rst file"""
+
+        heading_bar = "=" * len(display_name)
+        display_name_formatted = f"{heading_bar}\n{display_name}\n{heading_bar}"
+        return display_name_formatted
+
+    def render_doc(self, config_settings: ConfigSettings, config_step) -> None:
+        """
+        Render a `ConfigSettings` documentation template with the following variables:
+            1. enable_setting
+            2. required_settings
+            3. all_settings (required_settings + optional_settings)
+            4. detailed_info
+            5. title
+            6. link (for crossreference across different files)
+        """
+        # 1.
+        enable_setting = getattr(config_step, "enable_setting", None)
+
+        # 2.
+        required_settings = [
+            name for name in getattr(config_settings, "required_settings", [])
+        ]
+
+        # additional requirements from related configuration steps
+        related_steps = [step for step in getattr(config_step, "related_steps", [])]
```

Comment:
It's not required (hence the `getattr`), but it can be used to embed the documentation for several configuration steps in another one. For example, if you have a config step for `Foo`, which has two API services `Bar` and `Baz` associated with it, I don't want three documentation files `foo_doc.rst`, `bar_doc.rst`, `baz_doc.rst`. I want a single file that contains the documentation for everything in one place; otherwise the docs get cluttered and harder to process.

Example use case in OIP (the PR does not reflect the recent changes to the library): https://github.com/maykinmedia/open-inwoner/blob/1cb83262cfe7ef29e5816c96ebdb51b78bfa73d1/src/open_inwoner/configurations/bootstrap/zgw.py#L341

I've updated the comment to make the intention more clear. Let me know if this makes sense, otherwise I'll have to think of a different solution for our case in OIP.

Reply:
Thanks for explanation and the example.  I think of two options how to make it a little clearer:

1.  Declare `related_steps` variable to BaseConfigurationStep, it can have default as empty list. So it would be added to the explicit API of the class. 
2.  Am I correct that `ConfigurationStep.related_steps` variable is used only to generate documentation now? If this is true and `related_steps` is not planned to use in the logic of configuration steps, this variable could be moved to `ConfigSettings` class as `related_config_settings`

Anyway the explicit declaration of the variable with the typehint would be appreciated